### PR TITLE
Several fixes for Gemma3n

### PR DESCRIPTION
**Pull request by Cyrilvallez**
# What does this PR do?

### CODE REVIEW

File: src/transformers/models/gemma3n/modeling_gemma3n.py [hunk 1]
```diff
@@ -1656,18 +1664,17 @@ def forward(
         position_embeddings_local = self.rotary_emb_local(hidden_states_0, position_ids)
 
         # Expand hidden_states to support per-layer inputs
-        target_magnitude: torch.Tensor = torch.mean(hidden_states_0**2, dim=-1, keepdim=True) ** 0.5
-        epsilon_tensor = torch.tensor(torch.finfo().min)
+        target_magnitude = torch.mean(hidden_states_0**2, dim=-1, keepdim=True) ** 0.5
+        epsilon_tensor = torch.tensor(1e-5)
```
Comment (ArthurZucker) [inline src/transformers/models/gemma3n/modeling_gemma3n.py:1668]: Seems a little bit unrelated to me, where does that come from?

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

File: src/transformers/models/gemma3n/modeling_gemma3n.py (hunk 1 above)
Comment (Cyrilvallez) [reply src/transformers/models/gemma3n/modeling_gemma3n.py:1668]: Otherwise we can get `NaN` on those layers because of the line `current_hidden_state = current_hidden_state * (target_magnitude / torch.maximum(new_magnitude, epsilon_tensor))` - and the max ops is completely useless if we compare it to the minimum possible for any given dtype. Given the `epsilon` name of the variable, I assumed it is a typo and was supposed to be a small value instead for numerical stability

**Comment by ydshieh** [comment]
run-slow: gemma3n

**Comment by Cyrilvallez** [comment]
All good, @ydshieh for the slow tests! (IntegrationTests are still skipped, I will check them soon)

**Comment by danielhanchen** [comment]
Do you guys know why the training loss is exceptionally high? I don't think it's due to the gradient accumulation - it does quickly decrease, but it's very weird

**Comment by ArthurZucker** [comment]
👀 no idea!

**Comment by danielhanchen** [comment]
Wait I mis-spoke - grad accumulation does in fact not work correctly.

But the losses are still suspiciously high
